### PR TITLE
DAOS-14623 libfabric: enable psm3 provider

### PR DESCRIPTION
**Pull request by soumagne**

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -11,6 +11,7 @@ configure_providers = \
 	--enable-verbs \
 	--enable-rxm \
 	--enable-shm \
+	--enable-psm3 \
```
Comment (brianjmurrell) [inline debian/rules:14]:
```
[2023-11-14T17:17:06.444Z] configure:24882: checking for linux/idxd.h
[2023-11-14T17:17:06.444Z] configure:24882: result: no
[2023-11-14T17:17:06.444Z] configure:25070: psm3 provider: disabled
[2023-11-14T17:17:06.444Z] configure:25107: WARNING: psm3 provider was requested, but cannot be compiled
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Needs a `packaging/` update also.

File: debian/rules (hunk 1 above)
Comment (soumagne) [reply debian/rules:14]: ok I'm trying something now but I might want to keep it disabled on debian builds...

**Comment by soumagne** [comment]
Not pursuing that for now

**Pull request closed without merging**